dash/dash.py

```diff
@@ -419,6 +419,42 @@ def index(self, *args, **kwargs):  # pylint: disable=unused-argument
     def interpolate_index(self,
                           metas='', title='', css='', config='',
                           scripts='', app_entry='', favicon=''):
+        """
+        Called to create the initial HTML string that is loaded on page.
+        Override this method to provide you own custom HTML.
+
+        :Example:
+
+            class MyDash(dash.Dash):
+                def interpolate_index(self, **kwargs):
+                    return '''
+                    <!DOCTYPE html>
+                    <html>
+                        <head>
+                            <title>My App</title>
+                        </head>
+                        <body>
+                            <div id="custom-header">My custom header</div>
+                            {}
+                            {}
+                            {}
+                            <div id="custom-footer">My custom footer</div>
+                        </body>
+                    </html>
+                    '''.format(
+                        kwargs.get('app_entry'),
+                        kwargs.get('config'),
+                        kwargs.get('scripts'))
+
```

Comment:
I'd have a preference for using named interpolations here, as I think this is good practice for such a long string being interpolated. eg `'{scripts}'` ... `.format(scripts=kwargs['scripts'])` But it's not that much of an issue.